### DISCUSSION TITLE
Tuya WiFi Smart Meter with CT (pj-mgw1103)

### DISCUSSION THREAD

**Discussion by giovanne123** [arendst/Tasmota#18396] [Ideas]
Would it be possible to support the device?
(or is it already? But I can't find a tasmota config for it)
Tuya WiFi Smart Meter with CT 
Model No. pj-mgw1103

AliExpress: https://a.aliexpress.com/_EQTO1Mb
Tuya Smart WiFi Strom Energiezähler 80A mit Zange Stromwandler App Monitor Stromverbrauch KWh Stromstatistik Digital Electric 110V 230V 50/60Hz

I found some related infos in other activities like tuya local, ... which maybe are helpful for tasmota integration/config!?
https://github.com/make-all/tuya-local/blob/main/DEVICES.md
https://github.com/make-all/tuya-local/issues/604
https://github.com/make-all/tuya-local/issues/336
https://community.home-assistant.io/t/tuya-1-or-3-phase-energy-meter/351386/14

I think flashing with tuya-convert like described in following link could be successful (because device is for Tuya Smart App or  Smart Life App) but I haven't tested because of haven't found a tasmota config so far:
https://www.simon42.com/tuya-convert-tasmota-firmware-smart-life/
(I haven't connected the device with tuya app so far to prevent firmware update which maybe make it not possible to flash other fw)

Will continue researching, but maybe someone has already successful flashed tasmota with config in it ?

**Comment by barbudor** [comment]
Just in case you've been on another planet for the last 2..3 years:
- Tuya has blocked upload through tuya-convert.
- Tuya has stopped using ESP processors which are the only ones supported by Tasmota. So you can't run Tasmota on most recent Tuya product unless you replace the Wifi module by an ESP based module (de-soldering skills required)

So unless you have the device, opened and confirmed it has an ESP porcessor with reachable pins for serial flashing, just throw away your dreams and choose another device.

There are plenty of devices that are supported by Tasmota which can provide power measurment.
- If you are looking for using a current clamp in order to go over 20A, then consider the Shelly EM
- If you are ready for little bit of DIY, get a NodeMCU (or any ESP board) and a PZEM-004t
- If you are really looking for cheap, you can almost directly connect a current clamp to a NodeMCU as descrived here: https://tasmota.github.io/docs/ADC/#adc_ct-circuit

Beside, for your knowledge Tuya-Local is a software plugin for Home Assistant to directly deal with Tuya devices without the need to install Tasmota. It works by locally riunning a service which let the device believe it is connected to the Tuya cloud. So this is of no interrest for Tasmota users as our goal it got totally rid of Tuya software both in the device and in the home server

**Comment by giovanne123** [comment]
OK, I will go back to my planet...
Nevertheless thanks for the infos provided.

**Reply by profesor79** [reply]
OpenBeken landed, so whatever you have inside, there is a hope :D

**Comment by bilogic** [comment]
@giovanne123 why not take it apart and share some photos here? This item caught my eye too.  I guess what we need to see is just the chip and connections inside the box.

1. https://hittheroad.dev/howto-integrate-tuya-wifi-energy-meter-into-home-assistant-5739806378c5
I think this guy did it without flashing, but it's a big detour.

2. https://www.elektroda.com/rtvforum/topic3946128.html it's likely a CB2S inside

3. You can also try a firmware friendly with your CB2S https://github.com/openshwprojects/OpenBK7231T_App

**Comment by g6094199** [comment]
>     3. You can also try a firmware friendly with your CB2S https://github.com/openshwprojects/OpenBK7231T_App

...or a friendly community...
